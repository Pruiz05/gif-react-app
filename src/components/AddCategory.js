import React, {useState} from 'react'
import PropTypes from 'prop-types'

export const AddCategory = ({setCategories}) => {

     const [inputValue, setInputValue] = useState('')

     const handleInputChange = (e) =>{
        setInputValue(e.target.value)

        
     }

     const handleSubmit =(e)=>{
        e.preventDefault();

        if( inputValue.trim().length > 1){

           setCategories(c => [...c, inputValue])
           setInputValue('')
        }
      //   console.log('agregado');
      //   console.log(setCategories())
     }

    return (
        <form onSubmit={handleSubmit}>
           <input 
                type="text"
                value={inputValue}
                onChange={handleInputChange}/>
           {/* <button className="btn">Agregar</button> */}
        </form>
    )
}

// propiedad es requerida
AddCategory.propTypes = {
   setCategories: PropTypes.func.isRequired
}

