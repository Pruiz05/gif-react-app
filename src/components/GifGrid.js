import React, { useEffect, useState } from "react";
import PropTypes from "prop-types";

import { getGifs } from "../helpers/getGifs";
import useFetchGifs from "../hooks/useFetchGifs";
import { GifGridItem } from "./GifGridItem";

export const GifGrid = ({ category }) => {
  // const [images, setImages] = useState([]);

  const { data: images, loading } = useFetchGifs(category);

  // useEffect(() => {
  //   getGifs(category)
  //       .then(setImages)
  // }, [category]);

  return (
    <>
      <h3 className={"animate__animated animate__fadeIn"}> {category} </h3>

      {loading && <p className={"animate__animated animate__flash"}>Loading</p>}

      <div className={"card-grid"}>
        {images.map((img) => (
          //   return <li key={img.id}>{img.title}</li>;
          <GifGridItem key={img.id} {...img} />
        ))}
      </div>
    </>
  );
};

GifGrid.propTypes ={
  category: PropTypes.func.isRequired
}
