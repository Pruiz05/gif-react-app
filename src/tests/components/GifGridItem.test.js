import React from 'react';
import { shallow } from "enzyme"
import { GifGridItem } from "../../components/GifGridItem"

describe('<Pruebas en <GifGridItem />', () => {

    const title ='placeholder';
    const url = 'https://via.placeholder.com/150'



    test('debe de mostrar el componente correctamente', () => {
        const wrapper = shallow( <GifGridItem title={title} url={url}/>)
        expect(wrapper).toMatchSnapshot();
    })
    
})
