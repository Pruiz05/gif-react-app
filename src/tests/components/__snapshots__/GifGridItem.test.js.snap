// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pruebas en <GifGridItem /> debe de mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="placeholder"
    src="https://via.placeholder.com/150"
  />
  <p>
    placeholder
  </p>
</div>
`;
